manim/scene/scene.py

```diff
@@ -601,32 +601,39 @@ def get_moving_and_static_mobjects(self, animations):
         )
         return all_moving_mobject_families, static_mobjects
 
-    def compile_animations(self, *animations, **play_kwargs):
+    def compile_animations(self, *args, **kwargs):
         """
         Creates _MethodAnimations from any _AnimationBuilders and updates animation
         kwargs with kwargs passed to play().
-
         Parameters
         ----------
         *animations : Tuple[:class:`Animation`]
             Animations to be played.
-
         **play_kwargs
             Configuration for the call to play().
-
         Returns
         -------
         Tuple[:class:`Animation`]
             Animations to be played.
         """
-        for animation in animations:
-            if inspect.ismethod(animation):
+        animations = []
+        for arg in args:
+            if type(arg) == _AnimationBuilder:
```

Comment:
This should be checking `isinstance`, no?
```suggestion
            if isinstance(arg, _AnimationBuilder):
```

Reply:
why?

Reply:
Doesn't seem very pythonic to be checking type equality, particularly because that locks out any potential subclass of `_AnimationBuilder` from working when it could really work just fine. Part of pythonicity imo is giving developers lots of freedom/control (usually conveyed with the phrase "We're all consenting adults here"), and allowing subclasses gives more freedom and control.